CRM/Mailchimp/Sync.php

```diff
@@ -42,16 +42,16 @@ public function __construct($list_id) {
     $this->list_id = $list_id;
     $this->group_details = CRM_Mailchimp_Utils::getGroupsToSync($groupIDs=[], $list_id, $membership_only=FALSE);
     foreach ($this->group_details as $group_id => $group_details) {
-      if (empty($group_details['category_id'])) {
         $this->membership_group_id = $group_id;
-      }
     }
```

Comment:
End of foreach membership_group_id could either point to mailchimp-membership-group or interest-grouping.